### PR TITLE
[Onyx-707] Make TaskGroup executor thread to do encoding

### LINKED ISSUE #707: Make TaskGroup executor thread to do encoding/decoding
Curretly we have **separate** thread pools for TaskGroup execution and encoding/decoding. How about making TaskGroup executors to take responsibility of encoding/decoding?

There are several benefits:
* We have more concise and Netty-way code for lower data plane, as we don't have to maintain separate thread pool.
* The number of encoding/decoding threads is controlled by Tang NamedParameter.
    * We always have fixed number of threads **globally**. It does not scale to the number of TaskGroup threads for each executor.
    * It's increasingly harder to select appropriate numbers on various test scenarios.
* We don't have to care about controlling encoding/decoding threads in excpetional cases, like #706.

### PR DESCRIPTION
**Pull request by seojangho**
This PR downsizes codebase of the lower data plane while preserving its functionality by:

* Removing separate thread pooling and buffering for outbound operations(writeByte, writeBytes, writeFileArea, writeSerializedPartition).
* Removing related configuration parameters.
* Removing ClosableBlockingQueue class, which is now unnecessary.

Resolves #707

### CODE REVIEW

**Comment by sanha** [comment]
!rebuild

File: runtime/executor/src/main/java/edu/snu/onyx/runtime/executor/data/blocktransfer/BlockOutputStream.java [hunk 1]
```diff
@@ -326,96 +250,49 @@ private void checkWritableCondition() throws IOException {
...
     }
 
     /**
      * Writes a data frame from {@link ByteBuf}.
      *
+     * @param byteBuf     {@link ByteBuf} to write.
      * @param isLastFrame whether or not the frame is the last frame
      */
-    private void writeDataFrame(final boolean isLastFrame) {
+    private void writeDataFrame(final ByteBuf byteBuf, final boolean isLastFrame) {
```
Comment (sanha) [inline runtime/executor/src/main/java/edu/snu/onyx/runtime/executor/data/blocktransfer/BlockOutputStream.java]: Do we really need to maintain the method `writeDataFrame` and giving `null` for `byteBuf` if it is a last frame?

Comment (sanha) [reply runtime/executor/src/main/java/edu/snu/onyx/runtime/executor/data/blocktransfer/BlockOutputStream.java]: We can just make separate methods for non-last frame & last frame both.

**Comment by seojangho** [comment]
@sanha Thanks for the comment. Please check the updates.

**Pull request merged**